catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -293,6 +300,36 @@ public void testRouteCreationMissingMonitoredDirectory() throws Exception {
         empty());
   }
 
+  @Test
+  public void testInterruptedWhileWaitingForInputTransformers() throws InterruptedException {
```

Comment:
✏️ I would definitely like to see tests for conditions like:
- destroy called before init is done
- init timing out
- destroy called before init starts
- ...
The idea is to tests all possibilities of race conditions between the 3 threads involved